teamcity/resource_build_config.go

```diff
@@ -546,28 +569,50 @@ func expandParameterCollection(d *schema.ResourceData) (*api.Parameters, error)
 	return out, nil
 }
 
-func flattenParameters(dt *api.Parameters) (config map[string]string, sys map[string]string, env map[string]string) {
-	env, sys, config = make(map[string]string), make(map[string]string), make(map[string]string)
+func flattenParameters(dt *api.Parameters) (config map[string]string, sys map[string]string, env map[string]string,
```

Comment:
You only need to specify the return type once if they're all the same:

```
func flattenParameters(dt *api.Parameters) (config, sys, env, config_specs, env_specs map[string]string) {
}
```

Also, can you rearrange it so that the `config_specs` and `env_specs` are in the same order as `config` and `env`?

Also, what would be the additional work to also support specs for `sys_params` in the provider? (Just in the provider; we don't have to also support it in the module until we need it)

Reply:
Done